backend/battles/views.py

```diff
@@ -86,3 +88,33 @@ def get_context_data(self, **kwargs):  # noqa
             Q(user_creator=user) | Q(user_opponent=user), settled=False
         )
         return context
+
+
+class DetailBattleView(UserIsNotInThisBattleMixin, DetailView):
+    model = Battle
+    template_name = "battles/detail_battle.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        # Determine if you are opponent or creator
+        battle = context["battle"]
+        if self.request.user == battle.user_creator:
+            opponent = battle.user_opponent
+        else:
+            opponent = battle.user_creator
```

Comment:
How about you extract this to a helper?

Reply:
I meant something like `get_opponent` where you would receive the battle and the `request.user` and return it.